### PR TITLE
fix: change tsconfig resolution to allow list

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: src/deno_json/ts.rs [hunk 1]
```diff
@@ -96,86 +95,55 @@ impl Serialize for IgnoredCompilerOptions {
...
+  "strictFunctionTypes",
+  "strictNullChecks",
+  "strictPropertyInitialization",
+  "useUnknownInCatchVariables", // turn on by default
+  "types",
+  "checkJs",
+  "isolatedDeclarations",
+  "verbatimModuleSyntax",   // hook it up
+  "emitDecoratorMetadata",  // warn, unstable could be removed at any time
+  "experimentalDecorators", // warn, unstable could be removed at any time
```
Comment (lucacasonato) [inline src/deno_json/ts.rs:123]: We're not warning for these yet - or do you want to do that in CLI?

Comment (lucacasonato) [reply src/deno_json/ts.rs:123]: We're doing it in CLI.

**Review by lucacasonato — APPROVED** [review]
LGTM

File: src/deno_json/ts.rs (hunk 1 above)
Comment (dsherret) [reply src/deno_json/ts.rs]: Let's do it in the cli

**Pull request merged**